ProcessProcedureTests.swift

```diff
@@ -8,24 +8,224 @@
 import XCTest
 import ProcedureKit
 import TestingProcedureKit
+import Foundation
 @testable import ProcedureKitMac
 
 class ProcessProcedureTests: ProcedureKitTestCase {
 
     var process: Process!
     var processProcedure: ProcessProcedure!
 
+    var launchPath: String!
+    var arguments: [String]!
+
     override func setUp() {
         super.setUp()
 
-        process = Process()
-        process.launchPath = "/bin/echo"
-        process.arguments = [ "Hello World" ]
-        processProcedure = ProcessProcedure(process: process)
+        launchPath = "/bin/echo"
+        arguments = [ "Hello World" ]
+        processProcedure = ProcessProcedure(launchPath: launchPath, arguments: arguments)
     }
 
     func test__start_process() {
         wait(for: processProcedure)
         XCTAssertProcedureFinishedWithoutErrors(processProcedure)
     }
+
+    func test__start_process_with_launchpath_only() {
+        processProcedure = ProcessProcedure(launchPath: launchPath)
+        wait(for: processProcedure)
+        XCTAssertProcedureFinishedWithoutErrors(processProcedure)
+    }
+
+    func test__cancel_process_before_launched() {
+        processProcedure.cancel()
+        wait(for: processProcedure)
+        XCTAssertProcedureCancelledWithoutErrors(processProcedure)
+    }
+
+    func test__cancel_process_after_launched() {
+        processProcedure = ProcessProcedure(launchPath: "/bin/bash/", arguments: ["-c", "sleep 2"])
+        processProcedure.addDidExecuteBlockObserver { (procedure) in
+            procedure.cancel()
+        }
+        wait(for: processProcedure, withTimeout: 1)
```

Comment:
For future reference, per Slack discussion:

- `check` does not enforce timing between the execution of the block and the execution of the Procedure
- Thus, replacing the didExecuteObserver in this test with `check` can result in the cancellation sometimes occurring *prior* to the ProcessProcedure executing.
- Which won't result in test failures, but doesn't end up testing what we want in all cases.

PR #643 fixes this by adding a new `checkAfterDidExecute` method, and using it in this test.